### ISSUE TITLE
Image gallery not highlighting image (WEB)

### ISSUE THREAD

**Issue by esDotDev** [gskinnerTeam/flutter-wonderous-app#102]
The app creates a sem-transparent overlay with a hole/cutout in the middle, this is used to highlight the current image. 

It uses the `Path.combine` API with a `PathOperation.difference`, on web this is not working.

Ex code:
```
return Path.combine(
      PathOperation.difference,
      Path()..addRect(Rect.fromLTWH(0, 0, size.width, size.height)),
      Path()
        ..addRRect(
          RRect.fromLTRBR(
            padX,
            padY,
            size.width - padX,
            size.height - padY,
            Radius.circular(6),
          ),
        )
        ..close(),
```
Expected result: 
![image](https://github.com/gskinnerTeam/flutter-wonderous-app/assets/736973/434c92b6-56b4-40dd-bc46-788480685a6b)

Actual result: 
![image](https://github.com/gskinnerTeam/flutter-wonderous-app/assets/736973/dde96462-c011-4d07-933b-9ce6e812e8e2)

We do it this way as it makes it very easy to accomplish this effect that the designers requested, where it acts as an animated window/viewport over top of the images: 

https://github.com/gskinnerTeam/flutter-wonderous-app/assets/736973/3ac924fa-ddc5-4c82-a5a8-e474b1eb9225

**Comment by yjbanov** [comment]
Looks related to https://github.com/flutter/flutter/issues/124675

**Comment by esDotDev** [comment]
Oh perfect, ty!

**Comment by esDotDev** [comment]
btw, if this is not expected to be fixed any time soon, we could work around this by falling back to a less impressive effect on web. eg, we could just have use a semi-opaque overlay on each image, and fade it out on the selected image.

Wouldn't be quite as cool, but it would do the trick and side-step this issue for now.

**Comment by esDotDev** [comment]
@yjbanov Just curious if you think we can expect a fix in the short-term on this, or should we put a web-specific workaround in place for now?

**Comment by esDotDev** [comment]
Will do!

**Comment by esDotDev** [comment]
Closed in #152
